### PR TITLE
HDDS-5049. Add timeout support for ratis requests in SCM HA.

### PR DESCRIPTION
**Pull request by bshashikant**
## What changes were proposed in this pull request?
Added Timeout support for ratis requests.

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-5049

## How was this patch tested?

Existing UT

### CODE REVIEW

**Comment by bshashikant** [comment]
@GlenGeng , can you have a look?

File: hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/scm/ScmConfigKeys.java [hunk 1]
```diff
@@ -467,6 +467,11 @@
   public static final TimeDuration OZONE_SCM_RATIS_MINIMUM_TIMEOUT_DEFAULT
       = TimeDuration.valueOf(1, TimeUnit.SECONDS);
 
+  public static final String OZONE_SCM_RATIS_REQUEST_TIMEOUT_KEY
+      = "ozone.scm.ratis.minimum.timeout";
```
Comment (GlenGeng) [inline hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/scm/ScmConfigKeys.java:471]: Should we add the key to `SCMHAConfiguration` ? 
BTW, the key `OZONE_SCM_RATIS_SERVER_REQUEST_TIMEOUT_KEY`, `OZONE_SCM_RATIS_SERVER_RETRY_CACHE_TIMEOUT_KEY`, `OZONE_SCM_RATIS_MINIMUM_TIMEOUT_KEY`, `OZONE_SCM_LEADER_ELECTION_MINIMUM_TIMEOUT_DURATION_KEY` etc are not used any more. Better clean up them.

Comment (bshashikant) [reply hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/scm/ScmConfigKeys.java:471]: I think, we should be remove SCMHAConfiguration and pull all the required configs in ScmConfigKeys and remove the unsed ones. But, i would prefer to do this as a part of different jira altogether.

Comment (GlenGeng) [reply hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/scm/ScmConfigKeys.java:471]: Sure. Thanks for creating the jira.

**Comment by GlenGeng** [comment]
+1. Thanks @bshashikant for the work. I will merge it.

**Pull request merged**